src/openrct2-ui/scripting/ScWidget.hpp

```diff
@@ -69,40 +69,42 @@ namespace OpenRCT2::Scripting
             {
                 switch (widget->type)
                 {
-                    case WWT_FRAME:
+                    case WindowWidgetType::Frame:
                         return "frame";
-                    case WWT_RESIZE:
+                    case WindowWidgetType::Resize:
                         return "resize";
-                    case WWT_IMGBTN:
-                    case WWT_COLOURBTN:
-                    case WWT_TRNBTN:
-                    case WWT_FLATBTN:
-                    case WWT_BUTTON:
-                    case WWT_CLOSEBOX:
+                    case WindowWidgetType::ImgBtn:
+                    case WindowWidgetType::ColourBtn:
+                    case WindowWidgetType::TrnBtn:
+                    case WindowWidgetType::FlatBtn:
+                    case WindowWidgetType::Button:
+                    case WindowWidgetType::CloseBox:
                         return "button";
-                    case WWT_TAB:
+                    case WindowWidgetType::Tab:
                         return "tab";
-                    case WWT_LABEL_CENTRED:
-                    case WWT_LABEL:
+                    case WindowWidgetType::LabelCentred:
+                    case WindowWidgetType::Label:
                         return "label";
-                    case WWT_TABLE_HEADER:
+                    case WindowWidgetType::TableHeader:
                         return "table_header";
-                    case WWT_SPINNER:
+                    case WindowWidgetType::Spinner:
                         return "spinner";
-                    case WWT_DROPDOWN:
+                    case WindowWidgetType::DropdownMenu:
                         return "dropdown";
-                    case WWT_VIEWPORT:
+                    case WindowWidgetType::Viewport:
                         return "viewport";
-                    case WWT_GROUPBOX:
+                    case WindowWidgetType::Groupbox:
                         return "groupbox";
-                    case WWT_CAPTION:
+                    case WindowWidgetType::Caption:
                         return "caption";
-                    case WWT_SCROLL:
+                    case WindowWidgetType::Scroll:
                         return "scroll_view";
-                    case WWT_CHECKBOX:
+                    case WindowWidgetType::Checkbox:
                         return "checkbox";
-                    case WWT_TEXT_BOX:
+                    case WindowWidgetType::TextBox:
                         return "textbox";
+                    default:
+                        break;
```

Comment:
I think we shouldn't have a default in this case, as it's a plugin API (specially if we add a new widget type later, it's easy to forget. How many enumerations are missing here?

Reply:
That makes sense. It was "empty", "placeholder", and "last" left. I think these are sort of dummy values, but I added them and removed the default case.